includes/Media.php

```diff
@@ -273,4 +277,42 @@ public static function upload_mimes( array $mime_types ) {
 		$mime_types['svg'] = 'image/svg+xml';
 		return $mime_types;
 	}
+
+	/**
+	 * Deletes associated poster image when a video is deleted.
+	 *
+	 * This prevents the poster image from becoming an orphan because it is not
+	 * displayed anywhere in WordPress or the story editor.
+	 *
+	 * @param int $attachment_id ID of the attachment to be deleted.
+	 *
+	 * @return void
+	 */
+	public static function delete_video_poster( $attachment_id ) {
+		remove_action( 'pre_get_posts', [ __CLASS__, 'filter_poster_attachments' ] );
```

Comment:
Can't we just use `get_post_thumbnail_id` here?

Reply:
Not when the user has manually replaced the poster/thumbnail with something custom

Reply:
How will a custom poster/thumbnail have the post meta of `self::POSTER_POST_META_KEY`?

Reply:
Something like this?
```
	public static function delete_video_poster( $attachment_id ) {
		$poster_id = get_post_thumbnail_id( $attachment_id );
                 if( ! $poster_id ){
                    return;
                 }
		$is_poster = (bool) get_post_meta( $poster_id, self::POSTER_POST_META_KEY, true );
		if ( $is_poster ) {
			wp_delete_attachment( $post_id, true );
		}
	}
```

Reply:
That doesn't work when the user has previously replaced the generated poster with a custom one.